### PR TITLE
fix(sdk): new fix for multiple workers writing with gcsfuse bug [KFP SDK v2]

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
#8455 attempted to fix a FileExistsError thrown from within KFP SDK code that occurs when multiple workers are writing to the same file in a distributed training strategy.

The previous fix was a probabilistic fix that works most of the time, but not all of the time. This new fix should work all of the time when running pipelines on Vertex Pipelines.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/components/executor.py [hunk 1]
```diff
@@ -257,9 +257,17 @@ def _write_executor_output(self, func_output: Optional[Any] = None):
...
-        executor_output_path = self._input['outputs']['outputFile']
-        # This check is to reduce the likelihood that two or more workers (in a distributed training/compute strategy) attempt to write to the same executor output file at the same time using gcsfuse. Do not remove until fixed by gcsfuse.
-        if not os.path.exists(executor_output_path):
+        # This check is to ensure only one worker (in a mirrored, distributed training/compute strategy) attempts to write to the same executor output file at the same time using gcsfuse, which enforces immutability of files.
+        write_file = True
+
+        cluster_spec_string = os.environ.get('CLUSTER_SPEC')
+        if cluster_spec_string:
+            cluster_spec = json.loads(cluster_spec_string)
+            chief_node_label = 'workerpool0'
```
Comment (chensun) [inline sdk/python/kfp/components/executor.py:266]: nit: CONSTANT?

Comment (connor-mccarthy) [reply sdk/python/kfp/components/executor.py]: Done.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**